### PR TITLE
[sil] Add clarity to how load_weak [take] and store_weak effect an ob…

### PR DESCRIPTION
**Pull request by gottesmm**
…jects weak ref count.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: docs/SIL.rst [hunk 1]
```diff
@@ -2845,7 +2851,9 @@ Initializes or reassigns a weak reference.  The operand may be ``nil``.
 
 If ``[initialization]`` is given, the weak reference must currently either be
 uninitialized or destroyed.  If it is not given, the weak reference must
-currently be initialized.
+currently be initialized and after the evaluation the value that was originally
+referenced by the weak reference will have its weak reference count decremented
+by 1.
```
[the remark this answers is not in the document]
Comment (gottesmm) [reply docs/SIL.rst]: Good idea!

File: docs/SIL.rst [hunk 2]
```diff
@@ -2851,9 +2852,12 @@ Initializes or reassigns a weak reference.  The operand may be ``nil``.
...
 uninitialized or destroyed.  If it is not given, the weak reference must
-currently be initialized and after the evaluation the value that was originally
-referenced by the weak reference will have its weak reference count decremented
-by 1.
+currently be initialized. After the evaluation:
+
+* The value that was originally referenced by the weak reference will have
+  its weak reference count decremented by 1.
+* If the optional is non-nil, the reference in the optional's strong reference count is not
+  affected.
```
Comment (rjmccall) [inline docs/SIL.rst]: The "if" clause is unfortunate here; it ends up suggesting that the reference count is affected if the optional is nil, which is of course not what you mean.  Just "The new reference's weak reference is incremented, and its strong reference count is not affected." should be clear enough.  If you want to be explicit about what happens in the `nil` case, you should do that for all the places you're modifying.

**Review by rjmccall — APPROVED** [review]
WFM

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Pull request merged**